aiida/storage/psql_dos/migrator.py

```diff
@@ -70,6 +70,33 @@ class PsqlDosMigrator:
 
     def __init__(self, profile: Profile) -> None:
         self.profile = profile
+        self._engine = create_sqlalchemy_engine(self.profile.storage_config)
+        self._connection = None
+
+    def close(self) -> None:
+        """Close the connection if it was opened and dispose of the engine."""
+        if self._connection:
+            self._connection.close()
+            self._connection = None
+
+        self._engine.dispose()
+        self._engine = None
```

Comment:
If I understand correctly, this currently breaks subsequent calls to the `connection` property.

Perhaps the cleaner way would be to create `self._engine` not in the constructor, but lazily upon calling `self.property`.

Reply:
True. I was thinking that once you call `close` the `PsqlDosMigrator` instance should no longer be used. The same principle applies for the `RmqCommunicator` of `kiwipy` for example. But then again, I don't really see a downside to moving the creation of the engine to `self.connection` as well, so will move it there.